src/test/ui/traits/trait-param-without-lifetime-constraint.stderr

```diff
@@ -0,0 +1,18 @@
+error: `impl` item signature doesn't match `trait` item signature
+  --> $DIR/trait-param-without-lifetime-constraint.rs:14:5
+   |
+LL | pub trait HaveRelationship<To> {
+   |                            -- for `impl` items to implement the method, this type parameter might need a lifetime restriction like `To: 'a`
```

Comment:
this seems sort of misleading, no? 

That is, I think the proper fix here is

```rust
impl<'a> HaveRelationship<&'a ProofReader> for Article {
    fn get_relation(&self) -> &'a ProofReader {
    //~^ ERROR `impl` item signature doesn't match `trait` item signature
        &self.proof_reader
    }
}
```

but I'm not sure how the new error would lead me there....

Reply:
But that [doesn't work](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=35c96eeb7efc21a652520c1572720cf3) either because `HaveRelationship` needs a lifetime so that `get_relation` takes a `&'a self` instead of `&self`. Shouldn't the correct [change be](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=a9952985c9126cfdc588c0429f8961fa)?:

```rust
pub trait HaveRelationship<'a, To: 'a> {
    fn get_relation(&'a self) -> To;
}

impl<'a> HaveRelationship<'a, &'a ProofReader> for Article {
    fn get_relation(&'a self) -> &'a ProofReader {
        &self.proof_reader
    }
}
```

Reply:
Let's discuss this in the issue :)